### DISCUSSION TITLE
Is it possible to make Mattermost Team Edition use KanIDM for OIDC

### DISCUSSION THREAD

**Discussion by alexanderadam** [kanidm/kanidm#2711] [Q&A]
I found a lot of material to make [Mattermost Team Edition](https://hub.docker.com/r/mattermost/mattermost-team-edition/tags) run with [Keycloak](https://www.keycloak.org/) (like [this](https://github.com/mattermost/mattermost/issues/15606#issuecomment-1604461145), [this](https://devopstales.github.io/home/mattermost-keycloak-sso/) or [this](https://medium.com/@mrtcve/mattermost-teams-edition-replacing-gitlab-sso-with-keycloak-dabf13ebb99e)) but did anyone have documentation on how to make Kanidm work with Mattermost Team Edition as well? :eyes:

**Comment by Firstyear** [comment]
I don't think we have any documentation on mattermost specifically, but I don't see why it wouldn't work today :)

**Reply by alexanderadam** [reply]
Thank you for your reply.
The problem is that Mattermost Team Edition only really works with the  'GitLab OIDC configuration' out of the box. :roll_eyes: 

That's why people needed more configuration efforts to make it work with Keycloak.
I'm currently setting up new systems and it would be wonderful if I just could use KanIDM since it looks amazing so far.

That's why I wanted to ask in advance. :wink: 

**PS:** Thank you so much for your work! :pray:

**Reply by Firstyear** [reply]
Well it looks like https://devopstales.github.io/home/mattermost-keycloak-sso/ has the most helpful advice here: 

You would configure a client/rs in kanidm with: https://kanidm.github.io/kanidm/stable/integrations/oauth2.html#resource-server

Then you set:

```
nano /etc/mattermost/config.json
...
"GitLabSettings": {
    "Enable": false,
    "Secret": "<YOUR CLIENT BASIC SECRET HERE>",
    "Id": "<YOUR CLIENT ID HERE>",
    "Scope": "<LIST OF SCOPES HERE>",
    "AuthEndpoint": "https://idm.example.com/oauth2/authorise",
    "TokenEndpoint": "https://idm.example.com/oauth2/token",
    "UserApiEndpoint": "https://idm.example.com/oauth2/openid/<YOUR CLIENT ID HERE>/userinfo"
},
```
